modules/hosted-control-planes-troubleshooting.adoc

```diff
@@ -0,0 +1,109 @@
+
+* You must have the `hcp` command line interface installed. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/clusters/cluster_mce_overview#hosted-install-cli[Installing the hosted control planes command line interface].
+
+* You must have the OpenShift CLI (`oc`) installed.
+
+* You must ensure that the `kubeconfig` file is loaded and is pointing to the management cluster.
+
+.Procedure
+
+* To gather output for troubleshooting, enter the following commands:
++
+[source,terminal]
+----
+CLUSTERNAME="samplecluster"
+CLUSTERNS="clusters"
+
+mkdir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+hypershift dump cluster \
+    --name ${CLUSTERNAME} \
+    --namespace ${CLUSTERNS} \
+    --dump-guest-cluster \
+    --artifact-dir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+----
++
+.Example output
++
+[source,terminal]
+----
+2023-06-06T12:18:20+02:00   INFO    Archiving dump  {"command": "tar", "args": ["-cvzf", "hypershift-dump.tar.gz", "cluster-scoped-resources", "event-filter.html", "namespaces", "network_logs", "timestamp"]}
+2023-06-06T12:18:21+02:00   INFO    Successfully archived dump  {"duration": "1.519376292s"}
+----
+
+* To configure the command line interface so that it impersonates all of the queries against the management cluster by using a username or service account, enter the `hypershift dump cluster` command with the `--as` flag. 
++
+The service account must have enough permissions to query all of the objects from the namespaces, so the `cluster-admin` role is recommended to make sure you have enough permissions. The service account must be located or have permissions to query at least the `HostedControlPlane` namespace.
++
+If your username or service account does not have enough permissions, when you run the command, the output contains only the objects that you have permissions to access. During that process, you might see `forbidden` errors.
++
+** To use impersonation by using a service account, enter the following command, replacing values as necessary:
++
+[source,terminal]
+----
+CLUSTERNAME="samplecluster"
+CLUSTERNS="clusters"
+SA="samplesa"
+SA_NAMESPACE="default"
+
+mkdir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+hypershift dump cluster \
+    --name ${CLUSTERNAME} \
+    --namespace ${CLUSTERNS} \
+    --dump-guest-cluster \
+    --as "system:serviceaccount:${SA_NAMESPACE}:${SA}" \
+    --artifact-dir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+----
+
+** To use impersonation by using a username, enter the following command, replacing values as necessary:
++
+[source,terminal]
+----
+CLUSTERNAME="samplecluster"
+CLUSTERNS="clusters"
+CLUSTERUSER="cloud-admin"
+
+mkdir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+hypershift dump cluster \
+    --name ${CLUSTERNAME} \
+    --namespace ${CLUSTERNS} \
+    --dump-guest-cluster \
+    --as "${CLUSTERUSER}" \
+    --artifact-dir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+----
+
+
```

Comment:
Not critical, but I'd recommend making sure that your editor has the settings to remove trailing whitespace / trailing newlines